src/BlazorWebView/src/Maui/Android/BlazorWebViewHandler.Android.cs

```diff
@@ -142,5 +143,22 @@ internal IFileProvider CreateFileProvider(string contentRootDir)
 		{
 			return new AndroidMauiAssetFileProvider(Context.Assets, contentRootDir);
 		}
+
+		/// <summary>
+		/// Calls the specified <paramref name="workItem"/> asynchronously and passes in the scoped services available to Razor components.
+		/// </summary>
+		/// <param name="workItem">The action to call.</param>
+		/// <returns>Returns a <see cref="Task"/> representing <c>true</c> if the <paramref name="workItem"/> was called, or <c>false</c> if it was not called because Blazor is not currently running.</returns>
+		/// <exception cref="ArgumentNullException">Thrown if <paramref name="workItem"/> is <c>null</c>.</exception>
+		public virtual async Task<bool> TryDispatchAsync(Action<IServiceProvider> workItem)
+		{
+			ArgumentNullException.ThrowIfNull(workItem);
+			if (_webviewManager is null)
+			{
+				return false;
+			}
+
+			return await _webviewManager.TryDispatchAsync(workItem);
```

Comment:
Per the other thread's comments, I think this code is more idiomatic and has better exception behavior, so I prefer to keep it. I think the counter-arguments are just not as compelling.

Reply:
in this case, the only improvement would be to catch it in the virtual, which does not guarantee it since it can be overridden. I'm not a fan of making async  methods await if not needed due to the overhead, but I won't make it a blocker yet I don't see a major value added in this overloads.